File: Leetcode/1221. Split a String in Balanced Strings/solution.kt
class Solution {
    fun balancedStringSplit(s: String): Int {
        var count = 0
        var res = 0

        s.forEach {
            if (it == 'R') {
                count += 1
            } else {
                count -= 1
            }

            if (count == 0) {
                res += 1
            }
        }

        return res
    }
}

File: Leetcode/1221. Split a String in Balanced Strings/solution.rs
impl Solution {
    pub fn balanced_string_split(s: String) -> i32 {
        let mut count = 0;
        let mut res = 0;
        
        for c in s.chars() {
            if c == 'R' {
                count += 1;
            }
            else {
                count -= 1;
            }
            if count == 0 { res += 1; }
            
        }
        
        res
    }
}